sklearn/inspection/_partial_dependence.py

```diff
@@ -457,8 +498,19 @@ def partial_dependence(
         _get_column_indices(X, features), dtype=np.int32, order="C"
     ).ravel()
 
+    custom_values = custom_values or {}
+    if isinstance(features, (str, int)):
+        features = [features]
+    custom_values_idx = {
+        index: custom_values.get(feature)
+        for index, feature in enumerate(features)
+        if feature in custom_values
+    }
```

Comment:
I think it's fairly unclear that `custom_values_idx` maps from the sliced `X`. I think we can make it more clear by doing:

```python
X_subset = _safe_indexing(X, features_indices, axis=1)

custom_values = custom_values or {}
custom_values_for_X_subset = {...}
grid, values = _grid_from_X(X_subset, ...)
```